src/lib/cimbar_js/cimbar_recv_js.cpp

```diff
@@ -0,0 +1,219 @@
+/* This code is subject to the terms of the Mozilla Public License, v.2.0. http://mozilla.org/MPL/2.0/. */
+#include "cimbar_recv_js.h"
+
+#include "cimb_translator/Config.h"
+#include "encoder/Decoder.h"
+#include "encoder/escrow_buffer_writer.h"
+#include "extractor/Extractor.h"
+#include "fountain/fountain_decoder_sink.h"
+#include "serialize/str_join.h"
+#include "util/Timer.h"
+
+#include <opencv2/opencv.hpp>
+
+#include <algorithm>
+#include <memory>
+
+
+namespace {
+
+	std::shared_ptr<fountain_decoder_sink> _sink;
+	std::string _reporting;
+
+	cv::Mat _debugFrame;
+
+	TimeAccumulator _tScanExtract;
+	TimeAccumulator _tImgDecode;
+
+	// settings
+	unsigned _colorBits = 2;
+	int _modeVal = 68;
+
+	bool legacy_mode()
+	{
+		return _modeVal == 4;
+	}
+
+	cv::UMat get_rgb(void* imgdata, int width, int height, int type)
+	{
+		cv::UMat img;
+		switch (type)
+		{
+			case 12:
+			{
+				img = cv::Mat(height * 3/2, width, CV_8UC1, imgdata).getUMat(cv::ACCESS_RW).clone();
+				cv::cvtColor(img, img, cv::COLOR_YUV2RGB_NV12); // 12 or 21 :hmm:
+				return img;
+			}
+			case 420:
+			{
+				img = cv::Mat(height * 3/2, width, CV_8UC1, imgdata).getUMat(cv::ACCESS_RW).clone();
+				cv::cvtColor(img, img, cv::COLOR_YUV420p2RGB);
```

Comment:
Using firefox on linux, this does work for a claimed "I420" camera feed, but I'm not confident it's actually correct. Still investigating...

(worst case: we'll add a "just do the RGBA conversion up front" option in the UI)

Reply:
Was able to get a I420 out of an android phone that matches. Still wondering if the "force RGBA" option might be a good idea.